plugins/cilium-cni/types/types.go

```diff
@@ -20,20 +20,99 @@ import (
 	"io/ioutil"
 	"net"
 
-	ciliumv2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
-
 	cniTypes "github.com/containernetworking/cni/pkg/types"
 	"github.com/containernetworking/cni/pkg/types/current"
 	"github.com/containernetworking/cni/pkg/version"
 )
 
+// ENISpec is the ENI specification of a node.
+//
+// NOTE: This type was duplicated from pkg/k8s/apis/cilium.io/v2 to reduce
+// package dependencies. Any changes made here should be reflected in the
+// ENISpec type in pkg/k8s/apis/cilium.io/v2 as well.
```

Comment:
@tklauser what if someone does changes the ENISpec in `pkg/k8s/apis/cilium.io/v2`? Mind adding a comment there as well? It will be pretty confusing to swap it there and then expect that loading the CNI conf from disk to use those new fields added.

Reply:
Thanks, good point. Will add a note to `ENISpec` in `pkg/k8s/apis/cilium.io/v2` as well.